app/pagination.py

```diff
@@ -1,211 +1,262 @@
-                                 args=(*args, result_queue))
-            threads.append(t)
-            t.start()
-        for t in threads:
-            t.join()
-
+            thread = threading.Thread(
+                target=self._es.process_gen3_graphql_query, args=(*args, queue_)
+            )
+            threads_pool.append(thread)
+            thread.start()
+        for thread in threads_pool:
+            thread.join()
         result = {}
-        while not result_queue.empty():
-            data = result_queue.get()
+        while not queue_.empty():
+            data = queue_.get()
             result.update(data)
         return result
 
-    def handle_order_by_dataset_description_filter(self, filter):
+    def _handle_order_by_dataset_description(self, filter_):
+        """
+        Handler for updating submitter_id for order by dataset description
+        """
         result = {}
-        if "submitter_id" in filter:
+        if "submitter_id" in filter_:
             result["submitter_id"] = []
-            for dataset in filter["submitter_id"]:
+            for dataset in filter_["submitter_id"]:
                 result["submitter_id"].append(f"{dataset}-dataset_description")
         return result
 
-    def get_pagination_order(self, item):
-        filter_dict = self.handle_order_by_dataset_description_filter(
-            item.filter)
+    def _handle_pagination_order(self, item):
+        """
+        Handler for updating pagination data order
+        """
         query_item = GraphQLQueryItem(
-            node="pagination_order_by_dataset_description", limit=item.limit, page=item.page, filter=filter_dict, access=item.access, asc=item.asc, desc=item.desc)
+            node="pagination_order_by_dataset_description",
+            limit=item.limit,
+            page=item.page,
+            filter=self._handle_order_by_dataset_description(item.filter),
+            access=item.access,
+            asc=item.asc,
+            desc=item.desc,
+        )
         if "asc" in item.order:
             query_item.asc = "title"
         elif "desc" in item.order:
             query_item.desc = "title"
-        query_result = self.SGQLC.get_queried_result(query_item)
-
         # Include both public and private if have the access
-        ordered_dataset = []
-        for ele in query_result[query_item.node]:
+        ordered_datasets = []
+        for ele in self._es.process_gen3_graphql_query(query_item):
             dataset_id = ele["experiments"][0]["submitter_id"]
-            if dataset_id not in ordered_dataset:
-                ordered_dataset.append(dataset_id)
-        return ordered_dataset
+            if dataset_id not in ordered_datasets:
+                ordered_datasets.append(dataset_id)
+        return ordered_datasets
 
-    def get_pagination_data(self, item, match_pair, is_public_access_filtered):
+    def gain_pagination_data(self, item, match_pair, is_public_access_filtered):
```

Comment:
Is there a reason to rename this to gain? get is a more common term.

Reply:
It's been a while after changing it. Can't really remember the reason. I will change it back to "get".